src/components/CountryCode.tsx

```diff
@@ -0,0 +1,174 @@
+        suffixes: string[];
+    };
+    flags: {
+        png: string;
+    };
+}
+
+interface CountryListProps {
+    CCInputWidth: string;
+    onSetCountryCode: Dispatch<SetStateAction<string>>;
+}
+
+function CountryCodeList({ CCInputWidth, onSetCountryCode }: CountryListProps) {
+    const [countries, setCountries] = useState<CountryProps[]>([]);
+    const [isClicked, setIsClicked] = useState<boolean>(false);
+    const [selectedCountry, setSelectedCountry] = useState<string>("");
+    const [query, setQuery] = useState<string>("");
+    const inputRef = useRef<HTMLInputElement | null>(null);
+    const dropdownRef = useRef<HTMLDivElement | null>(null);
+
+    useEffect(() => {
+        if (isClicked && inputRef.current) {
+            inputRef.current.focus();
+        }
+    }, [isClicked]);
+
+    useEffect(() => {
+        async function getCountries() {
+            try {
+                const res = await fetch(`https://restcountries.com/v3.1/all`);
+                const data = await res.json();
+                setCountries(data);
+            } catch (error) {
+                console.log("data not fetched");
+            }
+        }
+        getCountries();
+    }, []);
+
+    useEffect(() => {
+        function handleClickOutside(event: MouseEvent) {
+            if (dropdownRef.current && !dropdownRef.current.contains(event.target as Node)) {
+                setIsClicked(false);
+            }
+        }
+        isClicked
+            ? document.addEventListener("mousedown", handleClickOutside)
+            : document.removeEventListener("mousedown", handleClickOutside);
+
+        return () => {
+            document.removeEventListener("mousedown", handleClickOutside);
+        };
+    }, [isClicked]);
+
+    const isSelectedCountry = countries.find((country) => country.name.common === selectedCountry);
+
+    const defaultCountry = countries.find((country) => country.name.common === "Nigeria");
+    useEffect(() => {
+        if (defaultCountry) {
+            onSetCountryCode(`${defaultCountry.idd?.root}${defaultCountry.idd?.suffixes?.[0]}`);
+        }
+    }, [defaultCountry, onSetCountryCode]);
+
+    const searchedCountries = countries.filter((country) =>
+        country.name.common.toLowerCase().includes(query.toLowerCase())
+    );
+
+    const getCountryCode = () => {
+        return isSelectedCountry
+            ? `${isSelectedCountry.idd?.root}${isSelectedCountry.idd?.suffixes?.[0]}`
+            : `${defaultCountry?.idd?.root}${defaultCountry?.idd?.suffixes?.[0]}`;
+    };
+
+    return (
+        <div
+            ref={dropdownRef}
+            className={`relative ${CCInputWidth} bg-[#f5f5f5] text-[#000000BF] font-[500] border border-[#0000001A] rounded-[8px] text-[0.75rem] xl:text-[0.85rem] h-[2.25rem] md:h-[2.85rem] lg:h-[3.125rem] pl-[1rem]`}
+        >
+            <button
+                type="button"
+                onClick={() => setIsClicked(!isClicked)}
+                onKeyDown={(e) => {
```

Comment:
There's no need for this event listener. Native Buttons already act this way when a click event is triggered